connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -390,7 +410,11 @@ public void run() {
                             log.trace("Finished read to end log for topic {}", topic);
                         } catch (TimeoutException e) {
                             log.warn("Timeout while reading log to end for topic '{}'. Retrying automatically. " +
-                                "This may occur when brokers are unavailable or unreachable. Reason: {}", topic, e.getMessage());
+                                     "This may occur when brokers are unavailable or unreachable. Reason: {}", topic, e.getMessage());
+                            continue;
+                        } catch (RetriableException | org.apache.kafka.connect.errors.RetriableException e) {
```

Comment:
`readToLogEnd` is called by `start`. Should we add similar exception handle for that?

Reply:
The previous `readToLogEnd()` that used the consumer did not have any special exception handling in the `start()` method. I added this here because retriable exceptions should not stop the KafkaBasedLog's thread.